include/chemfiles/selections/expr.hpp

```diff
@@ -25,6 +25,9 @@ class Selector {
     virtual std::string print(unsigned delta = 0) const = 0;
     /// Check if the `match` is valid in the given `frame`.
     virtual bool is_match(const Frame& frame, const Match& match) const = 0;
+    /// Clear any cached data. This must be called before using the selection
```

Comment:
Is it possible to cache a reference to the original Frame and call clear when a different frame is used. Just a thought.

Reply:
This would be hard, as references can not be rebound to a new object in C++. One could use std::reference_wrapper, or a raw pointer though. 

Do you have any use case in mind ?

Reply:
Nope, I was just going off that comment, It would be useful so that the user would not need to do this themself - no reason other than that.